### PR TITLE
Add onClick prop to page action group title

### PR DESCRIPTION
**Pull request by kemaltatar**
### WHY are these changes introduced?

Adding onClick handler for the title of the action group in `Page` component

https://user-images.githubusercontent.com/7462532/167627000-e40b3a74-ff56-44e4-a7c4-cb4423ffd1b2.mp4

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
<Page
      breadcrumbs={[{content: 'Products', url: '/products'}]}
      title="3/4 inch Leather pet collar"
      titleMetadata={<Badge status="success">Paid</Badge>}
      subtitle="Perfect for any pet"
      compactTitle
      primaryAction={{content: 'Save', disabled: true}}
      secondaryActions={[
        {
          content: 'Duplicate',
          accessibilityLabel: 'Secondary action label',
          onAction: () => alert('Duplicate action'),
        },
        {
          content: 'View on your store',
          onAction: () => alert('View on your store action'),
        },
      ]}
      actionGroups={[
        {
          title: 'Promote',
          onClick: (onOpen) => {
            alert('hey');
            onOpen();
          },
          actions: [
            {
              content: 'Share on Facebook',
              accessibilityLabel: 'Individual action label',
              onAction: () => alert('Share on Facebook action'),
            },
          ],
        },
      ]}
      pagination={{
        hasPrevious: true,
        hasNext: true,
      }}
    >
      <Card title="Credit card" sectioned>
        <p>Credit card information</p>
      </Card>
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, ping @ sarahill to update the Polaris UI kit

### CODE REVIEW

**Comment by alex-page** [comment]
@kemaltatar this might still be a work in progress but I would assume the actionlist does not open if someone overrides the `onClick` handler.

**Comment by kemaltatar** [comment]
Yes, I was assuming when the user provide an `onClick` handler they want to do something else before showing the action list. So they can call the `openActions` callback where ever they want.
For example, in our case we want to show a modal and not show the action list when the product status is draft.

File: polaris-react/UNRELEASED.md [hunk 1]
```diff
@@ -16,7 +16,8 @@ Use [the changelog guidelines](/documentation/Versioning%20and%20changelog.md) t
 - Pass `domId` as an argument to `onActiveOptionChange` prop on `Listbox` ([5627](https://github.com/Shopify/polaris/pull/5627))
 - Adding min-height to `Popover.Pane` when the `Popover` contains a `height` prop ([#5685](https://github.com/Shopify/polaris/pull/5685))
 - Adjusted a hardcoded `padding` value for `FileUpload` and replaced it with a spacing token ([#5675](https://github.com/Shopify/polaris/pull/5675))
-- Added disable property for the action groups in header [#5702](https://github.com/Shopify/polaris/pull/5702)
+- Added disable property to the action groups title of the Page header [#5702](https://github.com/Shopify/polaris/pull/5702)
```
Comment (kyledurand) [inline polaris-react/UNRELEASED.md]:
```suggestion
- Added `disable` prop to the action groups title of the Page header ([#5702](https://github.com/Shopify/polaris/pull/5702))
```

File: polaris-react/UNRELEASED.md [hunk 2]
```diff
@@ -16,7 +16,8 @@ Use [the changelog guidelines](/documentation/Versioning%20and%20changelog.md) t
...
+- Added onClick property to the action groups title of the Page header [#5751](https://github.com/Shopify/polaris/pull/5751)
```
Comment (kyledurand) [inline polaris-react/UNRELEASED.md]:
```suggestion
- Added `onClick` prop to the action groups title of the Page header ([#5751](https://github.com/Shopify/polaris/pull/5751))
```

**Review by kyledurand — APPROVED** [review]
Looks good to me. Just a couple of suggestions on the changelog to match our formatting

File: polaris-react/src/components/ActionMenu/components/MenuGroup/MenuGroup.tsx [hunk 3]
```diff
@@ -12,6 +12,8 @@ export interface MenuGroupProps extends MenuGroupDescriptor {
   accessibilityLabel?: string;
   /** Whether or not the menu is open */
   active?: boolean;
+  /** Callback when the menu is clicked */
+  onClick?(openActions: () => void): void;
```
Comment (AndrewMusgrave) [inline polaris-react/src/components/ActionMenu/components/MenuGroup/MenuGroup.tsx:16]: Would you mind updating the description on the PR stating the rationale behind the change?

From what I can tell, this api change isn't needed. If we provide a `onClick` it'll be called with `handleOpen`. But all `handleOpen`  does it call `onOpen` with the title. I may be missing something, but can you not just use the `onOpen` prop?

**Review by jorgenunezsiri — APPROVED** [review]

File: polaris-react/src/components/ActionMenu/components/MenuGroup/MenuGroup.tsx (hunk 3 above)
Comment (kemaltatar) [reply polaris-react/src/components/ActionMenu/components/MenuGroup/MenuGroup.tsx:16]: Currently there is no handler for the `actionGroup` title button click. `onOpen` is not a public prop for the `actionGroups` it's defined on the parent component

Comment (AndrewMusgrave) [reply polaris-react/src/components/ActionMenu/components/MenuGroup/MenuGroup.tsx:16]: Sorry, just assumed this was a public component. Ignore me 😄

**Pull request merged**